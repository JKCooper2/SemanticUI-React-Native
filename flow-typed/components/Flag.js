// @flow
type SizeType = 'tiny' | 'small' | 'medium' | 'large' | 'big';

export type FlagPropsType = {
    name: string;
    size: SizeType;
    shiny: boolean;
};