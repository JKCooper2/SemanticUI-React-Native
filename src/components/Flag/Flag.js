// @flow
import React, { PureComponent } from 'react';

import FlagIcon from 'react-native-flags';

import _ from 'lodash';

import type { FlagPropsType } from '~/types/components/Flag';

let SIZES = {
    tiny: 16,
    small: 24,
    medium: 32,
    large: 48,
    big: 64
};

class Flag extends PureComponent<FlagPropsType> {
    static defaultProps = {
        size: 'medium',
        shiny: false
    };

    render() {
        const { name, size, shiny } = this.props;

        let viewSize = _.get(SIZES, size, SIZES.medium);

        return (
            <FlagIcon
                code={name.toUpperCase()}
                size={viewSize}
                type={shiny ? 'shiny' : 'flat'}
            />
        );
    }
}

export default Flag;