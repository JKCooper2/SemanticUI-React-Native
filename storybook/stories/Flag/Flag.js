// @flow
import React, { Fragment } from 'react';
import { storiesOf } from '@storybook/react-native';

import Flag from '~/components/Flag';

const flagStories = storiesOf('Flag', module);

flagStories.add('Standard', () => <Flag name="au" />);

flagStories.add('Sizes', () => (
    <Fragment>
        <Flag name="au" size="tiny" />
        <Flag name="au" size="small" />
        <Flag name="au" size="medium" />
        <Flag name="au" size="large" />
        <Flag name="au" size="big" />
    </Fragment>
));

flagStories.add('Shiny', () => (
    <Fragment>
        <Flag name="au" />
        <Flag name="au" shiny />
    </Fragment>
));

export default flagStories;
