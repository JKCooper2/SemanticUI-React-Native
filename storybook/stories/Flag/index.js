// @flow
import Flag from './Flag';

export {
    Flag as default
};