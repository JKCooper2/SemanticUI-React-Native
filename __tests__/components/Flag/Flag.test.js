// @flow
import React from 'react';
import { shallow } from 'enzyme';

import Flag from '~/components/Flag';

test('renders correctly', () => {
    const flag = shallow(<Flag />);
    expect(flag).toMatchSnapshot();
});
