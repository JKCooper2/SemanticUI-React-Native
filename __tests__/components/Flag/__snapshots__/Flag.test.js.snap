// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Flag
  code=""
  size={32}
  type="flat"
/>
`;
